### PR TITLE
Reviews

### PR DESCRIPTION
**Pull request by rebekkaschaub**
- Add update and delete function for reviews in FE
- Add ReviewStats endpoint with calculate ReviewStats function
- Display ReviewStats on preview in resultslist and map popup

### CODE REVIEW

File: backend/src/main/java/de/neuefische/backend/service/ReviewService.java [hunk 1]
```diff
@@ -69,4 +74,17 @@ public Review updateReview(String reviewId, ReviewDto reviewDto) {
...
+    public ReviewStats calculateReviewStats(String id) {
+        List<Review> reviews = listAllReviewsOfCounselingCenter(id);
+        int count = reviews.size();
+        int sum = 0;
+        for (Review review : reviews) {
+            sum += review.getRating();
+        }
+        if(count==0){
+            return ReviewStats.builder().count(count).build();
+        }
```
Comment (jamarob) [inline backend/src/main/java/de/neuefische/backend/service/ReviewService.java:87]: Maybe move this up? No need to run the loop if this is 0.

File: frontend/src/components/CounselingCenterCard.js [hunk 2]
```diff
@@ -38,21 +40,22 @@ const Wrapper = styled.button`
...
+  grid-template-rows: 1fr min-content;
 
   img {
     width: 30px;
     height: 30px;
-    align-self: center;
+    justify-self: center;
+  }
+
+  > section {
```
Comment (jamarob) [inline frontend/src/components/CounselingCenterCard.js:53]: Which section does this target? ReviewStats? Wehen you expose a `className` prop on ReviewStats you can do like so with styled components
```
$(ReviewStasts){
  // some style rules
}
```

File: frontend/src/components/MyReviewCard.js [hunk 3]
```diff
@@ -2,11 +2,35 @@ import styled from "styled-components/macro";
...
 
 export default function MyReviewCard({ review }) {
+  const queryClient = useQueryClient();
+  const history = useHistory();
+  const { token } = useContext(AuthContext);
+
+  const deleteReview = useMutation(
+    () => {
+      return removeReview(token, review.reviewId);
+    },
```
Comment (jamarob) [inline frontend/src/components/MyReviewCard.js:19]: Imho implicit return is more readable.
```suggestion
    () => removeReview(token, review.reviewId),
```

File: frontend/src/components/MyReviewCard.js [hunk 4]
```diff
@@ -56,12 +76,18 @@ const Wrapper = styled.section`
     white-space: nowrap;
     margin-bottom: 0;
   }
+
+  p: first-child {
```
Comment (jamarob) [inline frontend/src/components/MyReviewCard.js:80]: This looks weird.. either `p:first-child` or `p :first-child`, depending on what you want to target.

File: frontend/src/components/ResultsMap.js [hunk 5]
```diff
@@ -85,9 +89,8 @@ const Wrapper = styled.div`
   position: absolute;
   top: 95px;
   left: 0;
-  //z-index: 5;
 
-  section {
+  div > section {
```
Comment (jamarob) [inline frontend/src/components/ResultsMap.js:93]: I don't see a section in the JSX above, would be more readable with a className or restyling of a styled component.

File: frontend/src/components/ReviewForm.js [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+  }
+
+  return (
+    <Form onSubmit={handleSubmit}>
+      <section>
+        <p>Gesamtbewertung</p>
+        <StarRating onChange={handleChange} value={review.rating} />
+      </section>
+      <label>
+        <p>Titel</p>
```
Comment (jamarob) [inline frontend/src/components/ReviewForm.js:31]: You can style the label, why do you use a paragraph here?

File: frontend/src/components/ReviewStats.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+const Wrapper = styled.section`
+  display: flex;
+  align-items: center;
+  p {
+    font-size: 13px;
+    margin: 0;
+  }
+
+  p:nth-child(3) {
```
Comment (jamarob) [inline frontend/src/components/ReviewStats.js:49]: Maybe use a className? Saves me from counting JSX elements 😉

**Review by jamarob — COMMENTED** [review]
Good job! 👍 left some comments

File: frontend/src/components/CounselingCenterCard.js (hunk 2 above)
Comment (rebekkaschaub) [reply frontend/src/components/CounselingCenterCard.js:53]: Somehow that doesn't work for me 😅
What exactly do you mean by expose a className prop on ReviewStats? And where do I have to put it?

**Pull request merged**